optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -406,13 +406,12 @@ protected void applyScoreDirectorFactoryProperties(IndexView indexView, SolverCo
         }
 
         if (solverConfig.getScoreDirectorFactoryConfig().getScoreDrlList() != null) {
-            boolean isDroolsDynamicPresent = isClassDefined("org.drools.dynamic.DynamicServiceRegistrySupplier");
-            if (!isDroolsDynamicPresent) {
+            boolean isKogitoExtensionPresent = isClassDefined("org.kie.kogito.quarkus.rules.deployment.RulesAssetsProcessor");
```

Comment:
What about using [`Capabilities`](https://quarkus.io/guides/writing-extensions#capabilities) to avoid an hard link to FQCN (in particular [`kogito-rules`](https://github.com/kiegroup/kogito-runtimes/blob/master/kogito-quarkus-parent/kogito-quarkus-rules-extension/kogito-quarkus-rules-deployment/src/main/java/org/kie/kogito/quarkus/rules/deployment/RulesAssetsProcessor.java#L29))?
```suggestion
            boolean isKogitoExtensionPresent = capabilities.isPresent("kogito-rules");
```